vllm/worker/model_runner.py

```diff
@@ -1264,11 +1264,15 @@ def prepare_model_input(
         """
         model_input = self._prepare_model_input_tensors(
             seq_group_metadata_list, finished_requests_ids)
-        sampling_metadata = SamplingMetadata.prepare(seq_group_metadata_list,
-                                                     model_input.seq_lens,
-                                                     model_input.query_lens,
-                                                     self.device,
-                                                     self.pin_memory)
+        if get_pp_group().is_last_rank:
```

Comment:
@andoorve I noticed this small optimization - only need to prepare the sampling metadata tensors in the last rank

Reply:
Thanks! There's a lot of these optimizations we didn't include last time. Just from what I can tell, I think this one in particular might not help E2E (will just increase the bubble size on the other workers) but doesn't hurt to have

Reply:
Yeah I figured that but doesn't harm to skip redundant work I guess

Reply:
Yup for sure